### ISSUE TITLE
Progress bar does not work in IntelliJ Idea Console

### ISSUE THREAD

**Issue by saidaspen** [ctongfei/progressbar#23]
When running an application using progress bar in IntelliJ Idea it behaves strangely.
![intellij-progressbar](https://user-images.githubusercontent.com/7727687/36340443-57f87c08-13dd-11e8-85bb-0caa7853bc7f.gif)

**Comment by ctongfei** [comment]
Does it work in terminal?  I haven't tried it under Windows -- it works for Mac.

Also, since you are using Consolas font, for better visual effects please set `ProgressBarStyle.ASCII` because the box-drawing glyphs are not aligned properly in Consolas.

**Comment by saidaspen** [comment]
Same behaviour with ASCII.

**Comment by ctongfei** [comment]
Does this problem only occurs in IntelliJ? Have you tried it outside of the IDE (e.g. using Windows `cmd`)?

**Comment by saidaspen** [comment]
Same example in three consoles:

cmd.exe
![cmd-progressbar](https://user-images.githubusercontent.com/7727687/36350627-dbc0ade4-149b-11e8-97f0-b97d73d3a287.gif)

git bash:
![gitbash-progressbar](https://user-images.githubusercontent.com/7727687/36350629-e234d236-149b-11e8-87c1-896e6b945fb1.gif)

IntelliJ Idea 2017.3 (Windows)
![intellij2-progressbar](https://user-images.githubusercontent.com/7727687/36350632-edb74512-149b-11e8-8d6f-973d12387231.gif)

**Comment by koppor** [comment]
I can confirm this issue.

IntelliJ IDEA 2017.3.3 (Ultimate Edition)
Build #IU-173.4301.25, built on January 15, 2018
Licensed to [JabRef](https://github.com/JabRef/jabref/) /  
Subscription is active until November 28, 2018
For non-commercial open source development only.
JRE: 1.8.0_152-release-1024-b11 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by albancolley** [comment]
The issue here is that TerminalFactory.get().GetWidth (ProgressThread.consoleWdith()) is returning a width of 1 for IntelliJ.   Adding a check for consoleWidth <= 10 and setting it to 80 (say) fixes the issue.

me.tongfei.progressbar.ProgressThread#consoleWidth
```java
   int consoleWidth() {
        Terminal terminal = TerminalFactory.get();

        int columnWidth = terminal.getWidth();
        columnWidth = columnWidth > 10? columnWidth: 80;
        
        return columnWidth;
    }
```

**Comment by saidaspen** [comment]
I like that approach actually, as a work-around at least.

**Comment by Veske** [comment]
I would open a ticket about this in Jetbrains' YouTrack also. They might just fix it for you.

**Comment by ctongfei** [comment]
Fixed in 0.6.0 using @albancolley 's workaround approach.
